libs/core/algorithms/include/hpx/parallel/algorithms/for_loop.hpp

```diff
@@ -1136,6 +1136,14 @@ namespace hpx::parallel { inline namespace v2 {
             // stride shall not be zero
             HPX_ASSERT(stride != 0);
 
+            if constexpr (std::is_integral<B>::value &&
+                std::is_signed<B>::value && std::is_integral<E>::value &&
+                std::is_signed<E>::value)
```

Comment:
```suggestion
            if constexpr (std::is_integral_v<B> && std::is_signed_v<B> && 
                std::is_integral_v<E> && std::is_signed_v<E>)
```

Reply:
`[proc] Executing command: /usr/local/bin/cmake --build /home/onepunchman/Desktop/for_loop_algo/build --config Debug --target my_hpx_program -j 8 --
[build] Consolidate compiler generated dependencies of target my_hpx_program
[build] [ 50%] Building CXX object CMakeFiles/my_hpx_program.dir/forloop.cpp.o
[build] In file included from /home/onepunchman/HPX_INSTALL/include/hpx/parallel/algorithm.hpp:54,
[build]                  from /home/onepunchman/HPX_INSTALL/include/hpx/local/algorithm.hpp:10,
[build]                  from /home/onepunchman/Desktop/for_loop_algo/forloop.cpp:7:
[build] /home/onepunchman/HPX_INSTALL/include/hpx/parallel/algorithms/for_loop.hpp: In function ‘typename hpx::parallel::util::detail::algorithm_result<ExPolicy, void>::type hpx::parallel::v2::detail::for_loop(ExPolicy&&, B, E, S, hpx::util::index_pack<Is ...>, Args&& ...)’:
[build] /home/onepunchman/HPX_INSTALL/include/hpx/parallel/algorithms/for_loop.hpp:1116:36: error: ‘is_integral_v’ is not a member of ‘std’; did you mean ‘is_integral’?
[build]  1116 |                 if constexpr (std::is_integral_v<B> && std::is_signed_v<B> &&
[build]       |                                    ^~~~~~~~~~~~~
[build]       |                                    is_integral
[build] /home/onepunchman/HPX_INSTALL/include/hpx/parallel/algorithms/for_loop.hpp:1116:51: error: expected primary-expression before ‘>’ token
[build]  1116 |                 if constexpr (std::is_integral_v<B> && std::is_signed_v<B> &&
[build]       |                                                   ^
[build] /home/onepunchman/HPX_INSTALL/include/hpx/parallel/algorithms/for_loop.hpp:1116:59: error: expected ‘)’ before ‘::’ token
[build]  1116 |                 if constexpr (std::is_integral_v<B> && std::is_signed_v<B> &&
[build]       |                              ~                            ^~
[build]       |                                                           )
[build] make[3]: *** [CMakeFiles/my_hpx_program.dir/build.make:76: CMakeFiles/my_hpx_program.dir/forloop.cpp.o] Error 1
[build] make[2]: *** [CMakeFiles/Makefile2:83: CMakeFiles/my_hpx_program.dir/all] Error 2
[build] make[1]: *** [CMakeFiles/Makefile2:90: CMakeFiles/my_hpx_program.dir/rule] Error 2
[build] make: *** [Makefile:124: my_hpx_program] Error 2
[build] Build finished with exit code 2`

@hkaiser  got the above error.
`is_integral_v’ is not a member of ‘std’; did you mean ‘is_integral’` this is the error I was getting , that's why I changed the function name. Even in rest of the code it is   `std::is_integral<B>::value`

Reply:
@deepaksuresh1411 do you have a C++17 compliant compiler (what version of the compiler do you use)? These features were added in [C++17](https://en.cppreference.com/w/cpp/types/is_integral). But that's odd because HPX does require C++17 by default.

Reply:
@hkaiser Now, it is working with `.._v` also.